src/scripting/lua_ui.cc

```diff
@@ -920,513 +1120,1270 @@ static std::function<void()> create_plugin_action_lambda(lua_State* L, const std
+	int32_t initial_state = get_table_int(L, "state", false);
+
+	UI::Align align = get_table_align(L, "align", false);
+	UI::Box::Resizing resizing = get_table_box_resizing(L, "resizing", false);
+
+	UI::Radiogroup* group = new UI::Radiogroup();
+
+	lua_getfield(L, -1, "buttons");
+	luaL_checktype(L, -1, LUA_TTABLE);
+	lua_pushnil(L);
+	while (lua_next(L, -2) != 0) {
+		std::string name = get_table_string(L, "name", true);
+		std::string icon = get_table_string(L, "icon", true);
+		std::string rtooltip = get_table_string(L, "tooltip", false);
+		int32_t rx = get_table_int(L, "x", false);
+		int32_t ry = get_table_int(L, "y", false);
+
+		UI::Radiobutton* radiobutton;
+		group->add_button(parent, UI::PanelStyle::kWui, name, Vector2i(rx, ry),
+		                  g_image_cache->get(icon), rtooltip, &radiobutton);
+
+		// Box layouting if applicable
+		if (as_box != nullptr) {
+			as_box->add(radiobutton, resizing, align);
+		}
+
+		lua_pop(L, 1);
+	}
+	lua_pop(L, 1);
+
+	group->set_state(initial_state, false);
+	if (std::string on_changed = get_table_string(L, "on_changed", false); !on_changed.empty()) {
+		group->changed.connect(create_plugin_action_lambda(L, on_changed));
+	}
+
+	group->manage_own_lifetime();
+}
+
+UI::Panel* LuaPanel::do_create_child_slider(lua_State* L, UI::Panel* parent) {
+	std::string name = get_table_string(L, "name", true);
+	unsigned orientation = get_table_button_box_orientation(L, "orientation", true);
+	int32_t val_min = get_table_int(L, "min", true);
+	int32_t val_max = get_table_int(L, "max", true);
+	int32_t val = get_table_int(L, "value", true);
+	uint32_t cursor_size = get_table_int(L, "cursor_size", false, 20);
+	bool dark = get_table_boolean(L, "dark", false);
+
+	if (val_min > val_max) {
+		report_error(L, "Malformed slider value range");
+	}
+	if (val < val_min || val > val_max) {
+		report_error(L, "Slider initial value out of range");
+	}
+
+	std::string tooltip = get_table_string(L, "tooltip", false);
+	int32_t x = get_table_int(L, "x", false);
+	int32_t y = get_table_int(L, "y", false);
+	int32_t w = get_table_int(L, "w", false);
+	int32_t h = get_table_int(L, "h", false);
+
+	UI::Slider* slider;
+	if (orientation == UI::Box::Vertical) {
+		slider = new UI::VerticalSlider(parent, name, x, y, w, h, val_min, val_max, val,
+		                                dark ? UI::SliderStyle::kWuiDark : UI::SliderStyle::kWuiLight,
+		                                cursor_size, tooltip);
+	} else {
+		slider = new UI::HorizontalSlider(
+		   parent, name, x, y, w, h, val_min, val_max, val,
+		   dark ? UI::SliderStyle::kWuiDark : UI::SliderStyle::kWuiLight, tooltip, cursor_size);
+	}
+
+	if (std::string on_changed = get_table_string(L, "on_changed", false); !on_changed.empty()) {
+		slider->changed.connect(create_plugin_action_lambda(L, on_changed));
+	}
+
+	return slider;
+}
+
+UI::Panel* LuaPanel::do_create_child_spinbox(lua_State* L, UI::Panel* parent) {
```

Comment:
Your example includes the `resizing` property. Is it possible to support it?

Reply:
That is implemented in `do_create_child` (line 1205) since this is a property of the parent box and not of the child panel.